### ISSUE TITLE
Sometimes switching to git clone does not work even though command says it was successful

### ISSUE THREAD

**Issue by nlighteneddesign** [joachim-n/drupal-project-contrib-development#3]
I have been using this for a couple of weeks and it has been great!
I'll premise this report with I jumped through hoops to break this, but wanted to report it in case others have the same issue or it's something you can fix.

I am working on https://www.drupal.org/project/eca_commerce/issues/3445586
Steps to reproduce:
Get drupal 10 site on ddev
Require eca_commerce (comes with all of the dependencies)
Use drupal-project-contrib-development to switch to the clone
Change to the branch in the issue I mentioned above.
Note that you can no longer run composer install since 2.x requires 10.3 and the drupal repo is 10.2 this is not a problem cause I was just updating code.
Change back to package install for eca commerce after committing
Work on another module
Try switching back to eca_commerce and receive the following message

```

Composer could not detect the root package (drupal/recommended-project) version, defaulting to '1.0.0'. See https://getcomposer.org/root-version

 Switching eca_commerce to a git clone...

 ! [NOTE] Git clone for eca_commerce already exists in the 'repos' directory.               

Composer could not detect the root package (drupal/recommended-project) version, defaulting to '1.0.0'. See https://getcomposer.org/root-version
Updating Composer to switch drupal/eca_commerce to the git clone path repository.
Composer could not detect the root package (drupal/recommended-project) version, defaulting to '1.0.0'. See https://getcomposer.org/root-version
Loading composer repositories with package information
Updating dependencies
Your requirements could not be resolved to an installable set of packages.

  Problem 1
    - Root composer.json requires drupal/eca_commerce ^1.0 -> satisfiable by drupal/eca_commerce[1.0.0].
    - drupal/eca_commerce 1.0.0 requires drupal/eca ^2.0 -> found drupal/eca[dev-2.0.x, 2.0.0-alpha1, 2.0.0-alpha2, 2.0.x-dev (alias of dev-2.0.x)] but the package is fixed to 1.1.5 (lock file version) by a partial update and that version does not match. Make sure you list it as an argument for the update command.

Use the option --with-all-dependencies (-W) to allow upgrades, downgrades and removals for packages currently locked to specific versions.
                                                                                            
 [OK] The project's composer.json is now using eca_commerce from the git clone. You can now 
      go to the git clone and switch to a feature branch. 
```

I was able to work around this by deleting the repos folder and the modules/contrib/eca_commerce folder
running composer install
running the clone switch.

**Comment by joachim-n** [comment]
Thanks for reporting! And great to hear you're finding it useful.

What branch was the clone of eca_commerce on at the time you tried to switch back to the clone?

The problem will be that that branch isn't compatible with the state of the rest of your project, and Composer can't make changes to a path repository.

So the quick fix would be to show a message telling the user to manually change to a compatible branch, and a more complex fix would to make that branch switch if we can.

**Comment by nlighteneddesign** [comment]
I was probably still on the issue branch in the repo, but I had switched back to the package and installed 1.x so that makes sense.

If I understand what you are saying correctly if I went back into the repo folder and checked out the 1.x branch and then used this tool to switch it would have worked.

**Comment by joachim-n** [comment]
Yes, if you put the repo to the 1.x branch it *should* work. I think Composer will treat 1.x as satisfying a ^1 requirement.

The guaranteed way to get it to work is to check out the 1.0.0 tag.
